### PR TITLE
fix: ensure the APM agent's own requests to APM server are not traced

### PR DESCRIPTION
**Pull request by trentm**
This fixes a couple of older issues and also ensures these internal
requests do not add noise to internal debug logging for instrumentation
async context tracking.

Fixes: #1168
Fixes: #1136

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by trentm** [comment]
### example of the test failing

when using `elastic-apm-http-client@9.8.1`

```
% node test/do-not-trace-self.test.js
TAP version 13
# APM server intake request should not be traced
ok 1 fixtures/do-not-trace-self.js did not error out
ok 2 APM server got an event metadata object
ok 3 APM server got a span event
ok 4 APM server got no other events
ok 5 APM server got a single request
ok 6 APM server req was a POST...
ok 7 ... to the intake API
not ok 8 APM server req had no traceparent header
  ---
    operator: equal
    expected: |-
      undefined
    actual: |-
      '00-8e9dddbb688414622828312540aed19f-780aa7a364034620-01'
    at: done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:33:11)
    stack: |-
      Error: APM server req had no traceparent header
          at Test.assert [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:304:54)
          at Test.bound [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at Test.strictEqual (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:468:10)
          at Test.bound [as equal] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:33:11)
          at ChildProcess.exithandler (child_process.js:299:7)
          at ChildProcess.emit (events.js:314:20)
          at maybeClose (internal/child_process.js:1022:16)
          at Socket.<anonymous> (internal/child_process.js:444:11)
          at Socket.emit (events.js:314:20)
  ...
not ok 9 APM server req had no tracestate header
  ---
    operator: equal
    expected: |-
      undefined
    actual: |-
      'es=s:1'
    at: done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:34:11)
    stack: |-
      Error: APM server req had no tracestate header
          at Test.assert [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:304:54)
          at Test.bound [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at Test.strictEqual (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:468:10)
          at Test.bound [as equal] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:34:11)
          at ChildProcess.exithandler (child_process.js:299:7)
          at ChildProcess.emit (events.js:314:20)
          at maybeClose (internal/child_process.js:1022:16)
          at Socket.<anonymous> (internal/child_process.js:444:11)
          at Socket.emit (events.js:314:20)
  ...
not ok 10 APM server req had no elastic-apm-traceparent header
  ---
    operator: equal
    expected: |-
      undefined
    actual: |-
      '00-8e9dddbb688414622828312540aed19f-780aa7a364034620-01'
    at: done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:35:11)
    stack: |-
      Error: APM server req had no elastic-apm-traceparent header
          at Test.assert [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:304:54)
          at Test.bound [as _assert] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at Test.strictEqual (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:468:10)
          at Test.bound [as equal] (/Users/trentm/el/apm-agent-nodejs12/node_modules/tape/lib/test.js:91:32)
          at done (/Users/trentm/el/apm-agent-nodejs12/test/do-not-trace-self.test.js:35:11)
          at ChildProcess.exithandler (child_process.js:299:7)
          at ChildProcess.emit (events.js:314:20)
          at maybeClose (internal/child_process.js:1022:16)
          at Socket.<anonymous> (internal/child_process.js:444:11)
          at Socket.emit (events.js:314:20)
  ...

1..10
# tests 10
# pass  7
# fail  3
```

### example of the test passing

when using `elastic-apm-http-client@9.9.0` (from https://github.com/elastic/apm-nodejs-http-client/pull/161)

```
% node test/do-not-trace-self.test.js
TAP version 13
# APM server intake request should not be traced
ok 1 fixtures/do-not-trace-self.js did not error out
ok 2 APM server got an event metadata object
ok 3 APM server got a span event
ok 4 APM server got no other events
ok 5 APM server got a single request
ok 6 APM server req was a POST...
ok 7 ... to the intake API
ok 8 APM server req had no traceparent header
ok 9 APM server req had no tracestate header
ok 10 APM server req had no elastic-apm-traceparent header

1..10
# tests 10
# pass  10

# ok
```

File: lib/instrumentation/http-shared.js [hunk 1]
```diff
@@ -169,13 +169,8 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
       var req = orig.apply(this, newArgs)
       if (!span) return req
 
-      if (getSafeHost(req) === agent._conf.serverHost) {
-        agent.logger.debug('ignore %s request to intake API %o', moduleName, { id: id })
```
Comment (trentm) [inline lib/instrumentation/http-shared.js:173]: NOTE: No longer need this guard.

File: package.json [hunk 2]
```diff
@@ -86,7 +86,7 @@
     "basic-auth": "^2.0.1",
     "cookie": "^0.4.0",
     "core-util-is": "^1.0.2",
-    "elastic-apm-http-client": "^9.8.1",
+    "elastic-apm-http-client": "^9.9.0",
```
Comment (trentm) [inline package.json:89]: NOTE: To get https://github.com/elastic/apm-nodejs-http-client/pull/161

File: test/_mock_apm_server.js [hunk 3]
```diff
@@ -5,6 +5,7 @@
 //    server.start(function (serverUrl) {
 //      // Test code using `serverUrl`...
 //      // - Events received on the intake API will be on `server.events`.
+//      // - Raw request data is on `server.requests`.
```
Comment (trentm) [inline test/_mock_apm_server.js:8]: NOTE: Added functionality to this mock for the test below.

**Review by trentm — COMMENTED** [review]
Notes for reviewers.

File: lib/instrumentation/http-shared.js (hunk 1 above)
Comment (astorm) [reply lib/instrumentation/http-shared.js:173]: Agreed -- just to say it out loud removing this guard clause does mean if a user has an application that

1. Was instrumented by the agent
2. Was making their own calls to APM Server

that they'll suddenly see spans for those requests when they wouldn't previously. 

This case is pretty far fetched though and "more span" isn't exactly a bad thing.

**Review by astorm — APPROVED** [review]
Changes to the mock look reasonable, and pulling in the new client to avoid instrumenting APM Server requests reasonable. Approving.

**Pull request merged**